cdci_data_analysis/configurer.py

```diff
@@ -33,89 +33,97 @@
 
 logger = logging.getLogger("conf")
 
-
-# TODO: this looks rather specific to INTEGRAL?
-
 class DataServerConf:
 
-    def __init__(self, data_server_url,
-                 data_server_remote_cache=None,
-                 dispatcher_mnt_point=None,
-                 dummy_cache=None,
-                 products_url=None,
-                 data_server_port=None):
-
-        if data_server_port is not None:
-            logger.warning(
-                "data_server_port is disregarded, since it is naturally included in the url")
-
-        logger.info("building config from %s %s %s", data_server_url,
-                    data_server_remote_cache, dispatcher_mnt_point)
+    def __init__(self, required_keys=None, allowed_keys=None, **kwargs):
+        if required_keys is None:
+            #temporary hardcode to preserve interface
+            required_keys = ['data_server_url', 'dummy_cache']
 
-        if data_server_url is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_url is None: {data_server_url}")
+        if allowed_keys is None:
+            #temporary hardcode to preserve interface
+            allowed_optional_keys = ['optional_key']
+        else:
+            allowed_optional_keys = [x for x in allowed_keys if x not in required_keys]
 
-        if data_server_remote_cache is None:
-            logger.warning(
-                f"problem constructing {self}: data_server_remote_cache is None: {data_server_remote_cache}")
+        conf = kwargs.copy()
 
-        if dispatcher_mnt_point is None:
-            logger.warning(
-                f"problem constructing {self}: dispatcher_mnt_point is None: {dispatcher_mnt_point}")
+        logger.info("building config from %s", conf)
 
-        self.data_server_url = data_server_url
 
-        # dummy prods local cache
-        self.dummy_cache = dummy_cache
+        try:
+            self.__setattr__('data_server_url', conf.pop('data_server_url'))
+            required_keys.remove('data_server_url')
+        except KeyError as e:
+            logger.error(
+                f"problem constructing {self}: data_server_url configuration key is required")
+            raise e
 
-        # path to dataserver cache
-        self.data_server_remote_path = data_server_remote_cache
+        # special cases (INTEGRAL specific)
+        # NOTE: these are required for integral, so need to be processed before for-cycle for the other required ones
+        if conf.get('data_server_remote_cache', None) is not None:
+            # path to dataserver cache
+            self.data_server_remote_path = conf.pop('data_server_remote_cache')
+        else:
+            self.data_server_remote_path = None
 
-        if dispatcher_mnt_point is not None:
-            self.dispatcher_mnt_point = os.path.abspath(dispatcher_mnt_point)
+        if conf.get('dispatcher_mnt_point', None) is not None:
```

Comment:
frankly I would take out the entire special treatment part in a separate function, and maybe get rid of it later when feasible. to make this clear and nice function more clear and nice

and also put it at the end. the two integral-specific keys are also allowed by optional, by interface-preserving standard

Reply:
It could be taken into a separate method. But it must be called in the constructor before other required keys, not at the end. the reason for this is that these keys should be pop-ed from dict to not being processed in general for-loop

Reply:
I agree, it is good to have it in a separate method, in order to get rid of it completely in the future